### PR TITLE
Bulk export partitioning fixes

### PR DESCRIPTION
**Pull request by jamesagnew**
* Bulk export batch jobs are now better aware of the partition context
* Remove 2 of the 3 redundant parameters objects for the bulk export batch job
* Adds a new pointcut that is called for each resource being added to a bulk export file

### CODE REVIEW

File: hapi-fhir-jpaserver-test-r4/src/test/java/ca/uhn/fhir/jpa/bulk/BulkDataExportTest.java [hunk 1]
```diff
@@ -700,18 +722,57 @@ public void testSystemBulkExport() {
...
 		JobInstance finalJobInstance = verifyBulkExportResults(options, expectedIds, List.of());
 		assertEquals(40, finalJobInstance.getCombinedRecordsProcessed());
 	}
 
-	private JobInstance verifyBulkExportResults(BulkDataExportOptions theOptions, List<String> theContainedList, List<String> theExcludedList) {
-		Batch2JobStartResponse startResponse = myJobRunner.startNewJob(mySrd, BulkExportUtils.createBulkExportJobParametersFromExportOptions(theOptions));
+	@Test
+	public void testSystemBulkExport_WithBulkExportInclusionInterceptor() {
+
+		class BoysOnlyInterceptor {
```
Comment (tadgh) [inline hapi-fhir-jpaserver-test-r4/src/test/java/ca/uhn/fhir/jpa/bulk/BulkDataExportTest.java:738]: opposite of https://www.youtube.com/watch?v=MGJZfmYMltM

File: hapi-fhir-test-utilities/src/main/java/ca/uhn/fhir/test/utilities/server/BaseJettyServerExtension.java [hunk 2]
```diff
@@ -173,6 +181,9 @@ public void onClosed(Connection connection) {
 		contextHandler.setContextPath(myContextPath);
 		contextHandler.addServlet(servletHolder, myServletPath);
 		contextHandler.addFilter(new FilterHolder(requestCapturingFilter()), "/*", EnumSet.allOf(DispatcherType.class));
+		for (Filter next : myServletFilters) {
+			contextHandler.addFilter(new FilterHolder(next), "/*", EnumSet.allOf(DispatcherType.class));
+		}
```
Comment (tadgh) [inline hapi-fhir-test-utilities/src/main/java/ca/uhn/fhir/test/utilities/server/BaseJettyServerExtension.java:186]: thought: huh.... how did we ever register the filters before?

**Review by tadgh — APPROVED** [review]

File: hapi-fhir-test-utilities/src/main/java/ca/uhn/fhir/test/utilities/server/BaseJettyServerExtension.java (hunk 2 above)
Comment (jamesagnew) [reply hapi-fhir-test-utilities/src/main/java/ca/uhn/fhir/test/utilities/server/BaseJettyServerExtension.java:186]: This is only for the JUnit extension.. We've just never needed to register a filter for one of those before. I'm only doing it here so I can use it to simulate Auth in a smile test

File: hapi-fhir-jpaserver-test-r4/src/test/java/ca/uhn/fhir/jpa/bulk/BulkDataExportTest.java (hunk 1 above)
Comment (jamesagnew) [reply hapi-fhir-jpaserver-test-r4/src/test/java/ca/uhn/fhir/jpa/bulk/BulkDataExportTest.java:738]: lol!

File: hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/6_8_0/5039-add-bulk-export-pointcut.yaml [hunk 3]
```diff
@@ -0,0 +1,6 @@
+---
+type: change
+issue: 5039
+title: "A new interceptor pointcut called `STORAGE_BULK_EXPORT_RESOURCE_INCLUSION` has been added. This
```
[the remark this answers is not in the document]
Comment (jamesagnew) [reply hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/6_8_0/5039-add-bulk-export-pointcut.yaml:4]: Done

File: hapi-fhir-base/src/main/java/ca/uhn/fhir/interceptor/api/Pointcut.java [hunk 4]
```diff
@@ -1101,10 +1101,47 @@ public enum Pointcut implements IPointcut {
...
+	 * the resource object and this modification will affect the copy that is stored in the
+	 * bulk export data file (but will not affect the original). Hook methods may also
+	 * return <code>false</code> in order to request that the resource be filtered
+	 * from the export.
+	 * <p>
+	 * Hooks may accept the following parameters:
+	 * </p>
+	 * <ul>
+	 * <li>
+	 * ca.uhn.fhir.jpa.bulk.export.api.BulkDataExportOptions - The details of the job being kicked off
```
[the remark this answers is not in the document]
Comment (jamesagnew) [reply hapi-fhir-base/src/main/java/ca/uhn/fhir/interceptor/api/Pointcut.java:1124]: Whoops! fixed

File: hapi-fhir-base/src/main/java/ca/uhn/fhir/interceptor/api/Pointcut.java [hunk 5]
```diff
@@ -1101,10 +1101,47 @@ public enum Pointcut implements IPointcut {
 	 */
 	STORAGE_INITIATE_BULK_EXPORT(
 		void.class,
-		"ca.uhn.fhir.rest.api.server.bulk.BulkDataExportOptions",
+		"ca.uhn.fhir.rest.api.server.bulk.BulkExportJobParameters",
```
[the remark this answers is not in the document]
Comment (jamesagnew) [reply hapi-fhir-base/src/main/java/ca/uhn/fhir/interceptor/api/Pointcut.java:1104]: Agreed - I'm gonna mention this on the release webinar

**Pull request merged**